### PR TITLE
DEV-1310 / Drop removals and sync between writers

### PR DESCRIPTION
**Pull request by ricopinazo**
### What changes were proposed in this pull request?
This PR drops deletions capability and starts sending edges to both partitions to avoid synchronization between partitions. This changes the semantics for GraphPartition interface. Instead of `addVertex`, `syncNewEdgeAdd`, and `syncExistingEdgeAdd`, the subclasses now need to override: `addLocalEdge`, `addOutgoingEdge`, and `addIncomingEdge`

### Why are the changes needed?
These changes are needed to improve performance

### Does this PR introduce any user-facing change? If yes is this documented?
Yes, it removes `deleteEdge` and `deleteVertex` from the API. The methods were removed and so were their scaladoc. References to these methods in the full doc were removed as well.

### How was this patch tested?
unit and integration tests and also `TwitterTest` setting the `ArrowPartition` in the `BaseRaphtoryAlgotest` among other integrations tests disabling multithreaded ingestion

### Are there any further changes required?
No.

### CODE REVIEW

File: core/src/main/scala/com/raphtory/internals/components/RaphtoryServiceImpl.scala [hunk 1]
```diff
@@ -118,14 +119,18 @@ class RaphtoryServiceImpl[F[_]](
...
+    }
+    val updateMsg     = protocol.GraphAlterations(req.graphId, Vector(protocol.GraphAlteration(req.update)))
+
     for {
       _ <- F.delay(logger.trace(s"Processing graph update ${req.update}"))
-      _ <- partitions(partitioner.getPartitionForId(req.update.srcId)).processUpdates(
-                   protocol.GraphAlterations(req.graphId, Vector(protocol.GraphAlteration(req.update)))
-           )
-
+      _ <- F.parSequenceN(dstPartitions.size)(dstPartitions.map(_.processUpdates(updateMsg)).toSeq)
```
Comment (fabianmurariu) [inline core/src/main/scala/com/raphtory/internals/components/RaphtoryServiceImpl.scala:131]: does `toSeq` makes sense here?

**Review by fabianmurariu — APPROVED** [review]
LGTM

**Review by fabianmurariu — APPROVED** [review]
LGTM, please add the comments onto the new GraphPartition and be descriptive about whatthey do :)

File: core/src/main/resources/protocol.proto [hunk 2]
```diff
@@ -176,7 +176,7 @@ service PartitionService {
   rpc destroyGraph(GraphInfo) returns (Status);
   rpc establishExecutor(Query) returns (Status);
   rpc processUpdates(GraphAlterations) returns (google.protobuf.Empty);
-  rpc processEffects(GraphAlterations) returns (google.protobuf.Empty);
+//  rpc processEffects(GraphAlterations) returns (google.protobuf.Empty);
```
Comment (miratepuffin) [inline core/src/main/resources/protocol.proto:179]: Can we just remove this?

File: core/src/main/scala/com/raphtory/internals/components/partition/PartitionServiceImpl.scala [hunk 3]
```diff
@@ -78,11 +78,11 @@ abstract class PartitionServiceImpl[F[_]](
       _      <- writer.processUpdates(req)
     } yield Empty()
 
-  override def processEffects(req: GraphAlterations): F[Empty] =
-    for {
-      writer <- graphs.get.map(graphs => graphs(req.graphId).data.writer)
-      _      <- writer.processEffects(req)
-    } yield Empty()
+//  override def processEffects(req: GraphAlterations): F[Empty] =
```
Comment (miratepuffin) [inline core/src/main/scala/com/raphtory/internals/components/partition/PartitionServiceImpl.scala:81]: Can we just delete this

**Review by miratepuffin — APPROVED** [review]

**Review by miratepuffin — APPROVED** [review]

**Pull request merged**